content/en/docs/setup/version-skew-policy.md

```diff
@@ -0,0 +1,127 @@
+---
+reviewers:
+- sig-api-machinery
+- sig-architecture
+- sig-cli
+- sig-cluster-lifecycle
+- sig-node
+- sig-release
+title: Kubernetes Version and Version Skew Support Policy
+content_template: templates/concept
+weight: 70
+---
+
+{{% capture overview %}}
+This document describes the maximum version skew supported between various Kubernetes components.
+Specific cluster deployment tools may place additional restrictions on version skew.
+{{% /capture %}}
+
+{{% capture body %}}
+
+## Supported versions
+
+Kubernetes versions are expressed as **x.y.z**,
+where **x** is the major version, **y** is the minor version, and **z** is the patch version, following [Semantic Versioning](http://semver.org/) terminology.
+For more information, see [Kubernetes Release Versioning](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/release/versioning.md#kubernetes-release-versioning)
+
+The Kubernetes project maintains release branches for the most recent three minor releases,
+and expects users to be running those versions in production.
+
+Applicable bug and security fixes may be backported to those three release branches, depending on severity and feasibility.
+Patch releases are cut from those branches at a regular cadence,
+or on an as-needed basis as determined by the [patch release manager](https://github.com/kubernetes/sig-release/blob/master/release-team/role-handbooks/patch-release-manager/README.md#release-timing).
+
+With minor releases happening approximately every three months,
+that means each minor release is maintained for approximately nine months.
+
+## Supported version skew
+
+### `kube-apiserver`
+
+In multi-server clusters, the newest and oldest `kube-apiserver` instances must be within one minor version.
+
+Example:
+
+* newest `kube-apiserver` is at `1.13`
+* other `kube-apiserver` instances are supported at `1.13` and `1.12`
+
+### `kubelet`
+
+`kubelet` must not be newer than `kube-apiserver`, and may be up to two minor versions older.
```

Comment:
How does this jive with the "we only support three branches" policy above? If we assume that folks are happily running the apiservers up to a couple of minor releases behind the latest release, then the version of their kubelet will be >3 release branches old. And if we patch that release branch to fix a bug or security issue, are we continuing to verify that kubelets two releases behind keep working (via skew tests)?

Reply:
> then the version of their kubelet will be >3 release branches old

Currently, they would be required to upgrade their kubelets to match the same level as their apiserver to be within supported versions

Reply:
@kubernetes/sig-testing do the release-blocking tests for patch releases on the release branches test with version-skewed kubelets back two versions? (e.g. before a 1.10.x release gets cut, do we have tests demonstrating a 1.8.x kubelet still works against a 1.10.x apiserver)?

Reply:
tl;dr AFAICT, we don't exercise (control-plane=1.n, kubelet=1.n-2) in any meaningful capacity

- https://testgrid.k8s.io/sig-release-master-kubectl-skew: the testgrid dashboard names are misleading, this only ever tests version skew of n+-1, and only for kubectl vs. cluster
- https://testgrid.k8s.io/sig-release-master-upgrade: only tests 1.n -> 1.n+1 path; there is limited downgrade testing but it covers the entire cluster downgrading 
- https://testgrid.k8s.io/sig-cluster-lifecycle-upgrade-skew: has some 1.n-2 -> 1.n testing, but it's gke, and has been continuously failing

Just to be _technically correct_ about this, there is a very brief window where a version skew of n-2 may be tested, but it's not the version about to be released.  Our upgrade/release-related jobs use `ci/stable1-3`, `ci/beta`, and `ci/latest` aliases instead of calling out version numbers, to save on job config churn.  We are currently in the brief period of code freeze where `ci/stable1`\~=1.12, `ci/beta`\~=1.13, `ci/latest`=\~1.14, so any `ci/stable1`->`ci/latest jobs` are skewing by 2 versions.  However, since we are in code freeze, we're fast-forwarding master rather than cherry picking from master, so 1.14 is effectively the same as 1.13.  Once we release, `ci/stable1`\~=1.13, and max tested version skew goes back to 1.

Reply:
> I believe we used to have at least some tests that tested node/control-plane skew of two releases, by upgrading the control plane twice (IIRC).

I thought so as well.

> running a cluster persistently with nodes that are 2 releases behind the control plane does have risks that are useful to point out:
> 
> 1. Kubelet versions 2 releases behind will need to be replaced by upgraded ones before the control plane can be upgraded again
> 2. Since even the control-plane version is likely behind the latest open-source release, the even-older Kubelet versions won't be actively maintained (e.g., security patched) by the open-source project.
>
> Our general recommendation should be to try to make kubelet and control-plane versions match in the steady state, once they are sufficiently confident that the control-plane won't be downgraded.

will add notes/recommendations to that effect